File: umeng/android/src/main/java/com/yinxianwei/umeng/UmengPlugin.java
package com.yinxianwei.umeng;

import com.getcapacitor.Bridge;
import com.getcapacitor.CapConfig;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginConfig;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.umeng.commonsdk.UMConfigure;

@CapacitorPlugin(name = "Umeng")
public class UmengPlugin extends Plugin {

    private Umeng implementation = new Umeng();

    static public void preInit(Bridge bridge) {
        CapConfig config = bridge.getConfig();
        PluginConfig pluginConfig = config.getPluginConfiguration("Umeng");
        String appKey = pluginConfig.getString("androidAppKey");
        String channel = pluginConfig.getString("androidChannel");
        UMConfigure.preInit(bridge.getContext(), appKey, channel);
    }
    @PluginMethod
    public void initWithOptions(PluginCall call) {
        String appKey = this.getConfig().getString("androidAppKey");
        String channel = this.getConfig().getString("androidChannel");
        UMConfigure.init(this.getContext(), appKey, channel, UMConfigure.DEVICE_TYPE_PHONE, null);
        JSObject ret = new JSObject();
        ret.put("data", true);
        call.resolve(ret);
    }
    @PluginMethod
    public void setLogEnabled(PluginCall call) {
        Boolean value = call.getBoolean("value");
        UMConfigure.setLogEnabled(value);
        JSObject ret = new JSObject();
        ret.put("data", true);
        call.resolve(ret);
    }
}

File: umeng/ios/Sources/UmengPlugin/UmengPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(UmengPlugin)
public class UmengPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "UmengPlugin"
    public let jsName = "Umeng"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "initWithOptions", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "setLogEnabled", returnType: CAPPluginReturnPromise),
    ]
    private let implementation = Umeng()

    @objc func initWithOptions(_ call: CAPPluginCall) {
        let appKey = self.getConfig().getString("iOSAppKey") ?? ""
        let channel = self.getConfig().getString("iOSChannel")
        UMConfigure.initWithAppkey(appKey, channel: channel)
        call.resolve(["data": true])
    }
    @objc func setLogEnabled(_ call: CAPPluginCall) {
        let value = call.getBool("value")
        UMConfigure.setLogEnabled(value ?? false);
    }
}
